### DISCUSSION TITLE
Error: Error decrypting message: Could not find valid self-signature in key ac93a73e2d1sscd3

### DISCUSSION THREAD

**Discussion by packcode** [openpgpjs/openpgpjs#1744] [Q&A] [answered]
- OpenPGP.js version: @5.11.1
- Affected platform (Node.js version): v16.17.0

- Trying to decrypt using an old private key which does not have a self-signature even with allowInsecureDecryptionWithSigningKeys: true getting error Could not find valid self-signature in key ac93a73e2d1sscd3
``` 
const { data: decryptedData } = await openpgp.decrypt({
        message: await openpgp.readMessage({ binaryMessage: encryptedData }),
        decryptionKeys: privateKey,
        verificationKeys: publicKey,
        config: {
          allowInsecureDecryptionWithSigningKeys: true,
        },
      });

PS: New to GPG encryption/decryption process any help will be much appreciated

**Comment by larabr** [comment]
Hi @packcode , are you sure there is no self signature, or are you basing that assumption on the error you get?
If you are not sure, could you share the public key associated with your private key (you can get it via `privateKey.toPublic().armor()`)?

**Reply by packcode** [reply, marked as answer]
Hi @larabr The issue was with the private key. The approach I followed is to force imported the private key into **gpg (GnuPG)** and exported a new private key with self signature.
